### ISSUE TITLE
🪲 If-pressed not working

### ISSUE THREAD

**Issue by boryanagoncharenko** [hedyorg/hedy#5955]
There are multiple scenarios in which the if-pressed statement causes failures:
Level 9
```
points = 0
letters = a, b, c, d, e
repeat 10 times
    letter = letters at random
    print 'Press the letter ' letter
    if letter is pressed
        print 'great'
        points = points + 1
    else
        print 'oops wrong letter'
        points = points - 2

print points
```

Level 17
```
# Make 2 lists
colors = ['red','yellow','green','blue']
simon_sequence = []
player_sequence = []

# Create a function that adds a color
define add_random_color:
    color = colors[random]
    add color to simon_sequence

# Create a function that shows the simon_sequence
define show_simon_sequence with level:
    for i in range 1 to level:
        print simon_sequence[i]
        sleep 1
        clear

# Create a function that creates the player_sequence
define get_answer with level:
    for i in range 1 to level:
        print 'What is color number ' i '?'
        print 'press (r)ed, (y)ellow, (g)reen or (b)lue'
        if r is pressed:
            answer = 'red'
        elif y is pressed:
            answer = 'yellow'
        elif g is pressed:
            answer = 'green'
        elif b is pressed:
            answer = 'blue'
        else:
            print 'press (r)ed, (y)ellow, (g)reen or (b)lue'
        add answer to player_sequence


# Set up
level = 1
game_over = False
print 'Welcome to Simon Says!'
sleep 1
clear

# The game
while game_over == False:
    print 'this is level ' level
    sleep 1
    clear
    player_sequence = []

    call add_random_color
    call show_simon_sequence with level
    call get_answer with level
    if player_sequence == simon_sequence:
        print 'Well done!'
        sleep 1
        level = level + 1
    else:
        print 'Helaas pindakaas'
        game_over = True
```

**Comment by Felienne** [comment]
Also dropping all old pressed issues here, to check if they still make sense:

* #3680
* #3701
* #3857
* #3856
* #5727
* #5946
* #5757

**Comment by boryanagoncharenko** [comment]
> Also dropping all old pressed issues here, to check if they still make sense. I closed all of them since I suspect they no longer make sense with the new code, but could you check @boryanagoncharenko?
> 
>     * [[UI idea] Adding interface button for pressed feature #3680](https://github.com/hedyorg/hedy/issues/3680)
> 
>     * [[CONTENT] Make pressed possible with lowercase special keys #3701](https://github.com/hedyorg/hedy/issues/3701)
> 
>     * [[BUG] Hedy also shows key presses when the program is not waiting for one #3857](https://github.com/hedyorg/hedy/issues/3857)
> 
>     * [[BUG] Change "Waiting for a button press..." into "Waiting for a key press..." when appropriate #3856](https://github.com/hedyorg/hedy/issues/3856)
> 
>     * [🪲 If key pressed could be assigned a multi-letter string #5727](https://github.com/hedyorg/hedy/issues/5727)
> 
>     * [💻 Show "Waiting for a button press.." under the output field #5946](https://github.com/hedyorg/hedy/issues/5946)
> 
>     * [🪲 pressed not highlighted when number is used as key to press #5757](https://github.com/hedyorg/hedy/issues/5757)

Thank you for issue bundle. I checked them all and I think there are 2 which should definitely be addressed - I reopened them and assigned them to myself.

I just want to point out that one of the issues you closed (#3680) is a feature request which you mentioned you want to see done in the future. I assume if you closed it, then it did not pass in the vision-level meeting of Hedy.
